beets/autotag/__init__.py

```diff
@@ -61,22 +61,70 @@ def apply_item_metadata(item, track_info):
 def apply_metadata(album_info, mapping):
     """Set the items' metadata to match an AlbumInfo object using a
     mapping from Items to TrackInfo objects.
+
+    :param album_info: object to match
+    :type album_info: :py:class:`hooks.AlbumInfo`
     """
+    # Maps item attribute to AlbumInfo key
+    itemattr_to_albuminfokey = {
+        'albumartist_sort': 'artist_sort',
+        'albumartist_credit': 'artist_credit',
+        'disctotal': 'mediums',
+        'mb_albumid': 'album_id',
+        'mb_albumartistid': 'artist_id',
+        'mb_releasegroupid': 'releasegroup_id',
+        'comp': 'va',  # Compilation flag
+        'albumtype': 'albumtype',
+        'label': 'label',
+        'asin': 'asin',
+        'catalognum': 'catalognum',
+        'script': 'script',
+        'language': 'language',
+        'country': 'country',
+        'albumstatus': 'albumstatus',
+        'albumdisambig': 'albumdisambig',
+        'data_source': 'data_source',
+        'album': 'album',
+        'albumartist': 'artist',
+    }
+    # Maps item attributes to the corresponding TrackInfo key
+    itemattr_to_trackinfokey = {
+        'title': 'title',
+        'disc': 'medium',
+        'mb_trackid': 'track_id',
+        'track_alt': 'track_alt',
+        'lyricist': 'lyricist',
+        'composer': 'composer',
+        'composer_sort': 'composer_sort',
+        'arranger': 'arranger',
+        'media': 'media',
+        'disctitle': 'disctitle',
+    }
+    # Maps the item attribute to the corresponding key to be tried
+    # first on TrackInfo and then on AlbumInfo
+    itemattr_to_torainfokey = {
+        'artist': 'artist',
+        'artist_sort': 'artist_sort',
+        'artist_credit': 'artist_credit',
+        'mb_artistid': 'artist_id',
+    }
```

Comment:
Why not just make a loop over all tags? Then when one adds a tag he doesn't need to add something here.